src/charm.py

```diff
@@ -174,17 +138,23 @@ def __init__(self, *args):
         self.ingress_per_appv2 = ipa_v2 = IPAv2(charm=self)
 
         self.ingress_per_unit = IngressPerUnitProvider(charm=self)
+
         self.traefik_route = TraefikRouteProvider(
-            charm=self, external_host=self.external_host, scheme=self._scheme  # type: ignore
+            charm=self, external_host=self._external_host, scheme=self._scheme  # type: ignore
+        )
+
+        self.traefik = Traefik(
+            _TRAEFIK_SERVICE_NAME,
+            self.container,
+            routing_mode=self._routing_mode,
+            tcp_entrypoints=self._tcp_entrypoints(),
+            tls_enabled=self._is_tls_enabled(),
```

Comment:
`_on_cert_changed` calls `_configure_traefik` calls `Traefik.configure` calls `Traefik._update_tls_configuration()`.

So that path should be covered. Assuming that if cert-changed is firing, `Traefik._tls_enabled` (aka `cert_handler.enabled`) is True, traefik will dump the tls certs.